### PR TITLE
Database Bugfix #0037504: Return value of ilDBAnalyzer::getPrimaryKeyInformation() must be of the type array, bool returned

### PR DESCRIPTION
**Pull request by fhelfer**
https://mantis.ilias.de/view.php?id=37504

### CODE REVIEW

**Comment by mjansenDatabay** [comment]
@fhelfer I guess this has to be picked to `release_8` as well?

**Comment by fhelfer** [comment]
> @fhelfer I guess this has to be picked to `release_8` as well?

Indeed

File: Services/Database/classes/class.ilDBAnalyzer.php [hunk 1]
```diff
@@ -295,7 +295,7 @@ public function hasSequence(string $a_table)
         if (is_array($seq) && in_array($a_table, $seq)) {
             // sequence field is (only) primary key field of table
             $pk = $this->getPrimaryKeyInformation($a_table);
-            if (is_array($pk["fields"]) && count($pk["fields"]) === 1) {
+            if (isset($pk['fields']) & is_array($pk["fields"]) && count($pk["fields"]) === 1) {
```
Comment (chfsx) [inline Services/Database/classes/class.ilDBAnalyzer.php]: here one `&` is missing

**Comment by chfsx** [comment]
I merge this for trunk only.  please note that the ilDBAnalyzer is an internal class and can NOT be understood as a public API, so it should not be used in plugins or anything. the class can change or be dropped at any time.

**Pull request merged**